Frontend/src/hooks/useUserData.ts

```diff
@@ -0,0 +1,14 @@
+import { useQuery } from '@tanstack/react-query'
+import { getCompetitorsInfo } from '../api/user/post/get_user_info'
+
+export function useUserData(ids: number[]) {
+  return useQuery({
+    queryFn: () => (ids.length > 0 ? getCompetitorsInfo(ids) : []),
```

Comment:
I think `getCompetitorsInfo` should be able to deal with empty arrays without much hassle.

Reply:
After tracing the call, it seems like passing an empty array here to `getCompetitorsInfo` would post an empty array to the internal `api/v1`, which goes to the Rails `user_controller#info`. And in the body of that method, the (potentially empty!) IDs are being passed to `RedisHelper.cache_info_by_ids`.

IMHO, that `cache_info_by_ids` method should be able to deal with empty lists and just casually give back an empty list.